src/lib/tuya.ts

```diff
@@ -811,52 +811,54 @@ export const valueConverter = {
             return payload;
         },
     },
-    thermostatScheduleDayMultiDP: {
-        from: (v: string) => {
-            const schedule = [];
-            for (let index = 1; index < 17; index = index + 4) {
-                schedule.push(
-                    String(parseInt(v[index+0])).padStart(2, '0') + ':' +
-                    String(parseInt(v[index+1])).padStart(2, '0') + '/' +
-                    // @ts-ignore
-                    (parseFloat((v[index+2] << 8) + v[index+3]) / 10.0).toFixed(1),
-                );
-            }
-            return schedule.join(' ');
-        },
-        to: (v: string) => {
-            const payload = [0];
-            const transitions = v.split(' ');
-            if (transitions.length != 4) {
-                throw new Error('Invalid schedule: there should be 4 transitions');
-            }
-            for (const transition of transitions) {
-                const timeTemp = transition.split('/');
-                if (timeTemp.length != 2) {
-                    throw new Error('Invalid schedule: wrong transition format: ' + transition);
+    thermostatScheduleDayMultiDPN: (numberOfSchedules: number) => {
+        return {
+            from: (v: string) => {
+                const schedule = [];
+                for (let index = 1; index < 17; index = index + 4) {
```

Comment:
numberOfSchedules should be also used in this loop

index < 4*numberOfSchedules + 1 (this +1 actually make no difference ;-)

Reply:
```suggestion
                for (let index = 1; index < 4*numberOfSchedules + 1; index = index + 4) {
```

good catch